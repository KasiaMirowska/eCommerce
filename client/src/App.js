import React, { useEffect, lazy, Suspense } from 'react';
import { GlobalStyle } from './global.styles';

import { Route, Switch, Redirect } from 'react-router-dom';

import Header from './components/Header/Header';
import Spinner from './components/Spinner/Spinner';
import ErrorBoundary from './components/error-boundary/error-boundary';


const HomePage = lazy(() => import('./pages/homepage/HomePage'));
const ShopPage = lazy(() => import('./pages/shop/ShopPage'));
const LoginRegister = lazy(() => import('./pages/LoginRegister/LoginRegister'));
const CheckOut = lazy(() => import('./pages/checkOut/CheckOut'));

const App = (props) => {
  const {checkUserSession, currentUser} = props;

  useEffect(() => {
    checkUserSession()
  }, [checkUserSession]) //need to pass second argument- array in order to make effect invoke only once when mounting rather then every time state changes and rerenders.

  return (
    <div>
      <GlobalStyle />
      <Header />

      <Switch>
        <ErrorBoundary>
          <Suspense fallback={<Spinner />} >
            <Route exact path='/' component={HomePage} />
            <Route path='/shop' component={ShopPage} />
            <Route exact path='/signin' render={() => currentUser ? (<Redirect to='/' />) : (<LoginRegister />)} />
            <Route exact path='/checkout' component={CheckOut} />
          </Suspense>
        </ErrorBoundary>
      </Switch>
    </div>
  );
}



export default App;

