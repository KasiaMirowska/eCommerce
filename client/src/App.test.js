import React from 'react';
import { shallow } from 'enzyme';
import App from './App';


describe('App', () => {
  const props = {
    checkUserSession: jest.fn(),
    currentUser: 'user1'
  }

  const appWrapper = shallow(<App  {...props} />);

  it('renders as expected2', () => {
    expect(appWrapper).toMatchSnapshot();
  })

})

