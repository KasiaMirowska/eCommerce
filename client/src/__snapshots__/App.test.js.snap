// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders as expected2 1`] = `ShallowWrapper {}`;
